gradio/components.py

```diff
@@ -667,15 +674,41 @@ def _round_to_precision(num: float | int, precision: int | None) -> float | int:
         else:
             return round(num, precision)
 
+    @staticmethod
+    def _handle_min_max(
+        num: float | int, min: float | int, max: float | int
+    ) -> float | int:
+        """
+        If the number is greater than max, it returns max. If the number is less than min returns min. If the number is between min and max it returns the original number.
+
+        Args:
+            num (float | int): The input number to be capped.
+            min (int | float): The minimum allowed value.
+            max (int | float): The maximum allowed value.
+
+        Returns:
+            The input `num` capped between `min` and `max`.
+        """
+        if num > max:
+            return max
+        elif num < min:
+            return min
+        else:
+            return num
+
     def get_config(self):
         return {
             "value": self.value,
+            "max": self.max,
+            "min": self.min,
             **IOComponent.get_config(self),
         }
 
     @staticmethod
     def update(
         value: float | Literal[_Keywords.NO_VALUE] | None = _Keywords.NO_VALUE,
+        max: float | int = float("-inf"),
+        min: float | int = float("-inf"),
```

Comment:
I believe these should be `None`. Otherwise, calling `.update()` will reset the min/max:
 ```suggestion
        max: float | int | None = None,
        min: float | int | None = None,
```